conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -224,23 +224,36 @@ def update_autoreconf_args(self, updated_flags):
 
     # FIXME: Remove all these update_xxxx whenever xxxx_args are dicts or new ones replace them
     def _update_flags(self, attr_name, updated_flags):
-        _new_flags = []
-        self_args = getattr(self, attr_name)
-        for index, flag in enumerate(self_args):
-            flag_name = flag.split("=")[0]
-            if flag_name in updated_flags:
-                new_flag_value = updated_flags[flag_name]
-                # if {"build": None} is passed, then "--build=xxxx" will be pruned
-                if new_flag_value is None:
-                    continue
-                elif not new_flag_value:
-                    _new_flags.append(flag_name)
+
+        def _list_to_dict(flags):
+            ret = {}
+            for flag in flags:
+                # Only splitting if "=" is there
+                option = flag.split("=")
```

Comment:
```suggestion
                option = flag.split("=", 1)
```
Avoid potential strings with the "=" inside (unusual, but who knows)